src/Compiler/AST/Visitor/ReflectionAnalyzer.cpp

```diff
@@ -148,21 +454,40 @@ IMPLEMENT_VISIT_PROC(BufferDeclStmnt)
                     data_->textures.push_back(bindingSlot);
                 else
                     data_->storageBuffers.push_back(bindingSlot);
+
+                //ref uniform
+                Reflection::Uniform uniform;
+                uniform.ident = bufferDecl->ident;
+                uniform.type = Reflection::UniformType::Buffer;
+                uniform.baseType = (int)BufferTypeToReflType(ast->typeDenoter->bufferType);
+
+                //push
+                data_->uniforms.push_back(uniform);
             }
         }
     }
 }
 
 IMPLEMENT_VISIT_PROC(VarDecl)
-{
+{    
     if (ast->flags(AST::isReachable))
     {
         if (auto typeSpecifier = ast->FetchTypeSpecifier())
         {
             if (typeSpecifier->isUniform)
             {
+                DataType baseType = DataType::Undefined;
                 /* Add variable as uniform */
-                data_->uniforms.push_back(ast->ident);
+                Reflection::Uniform uniform;
+                uniform.ident = ast->ident;
+                uniform.type = Reflection::UniformType::Variable;
+                //find type
+                if (auto basetype_denoter = typeSpecifier->typeDenoter->As<BaseTypeDenoter>())
+                    uniform.baseType = (int)DataTypeToReflType(basetype_denoter->dataType);
```

Comment:
Please be so good and always use the appropriate C++ type cast. I never use C casts in my C++ projects for several reasons:
1. They can silently cast constness away! This is really evil and I have only exactly three locations where I use `const_cast` (see Expr::Find` and co) which is documented why it is ok at that point.
2. You cannot always know exactly which kind of cast the compiler will use, since C-style casts are a combination of multiple casts but only one is choosen by the compiler.
3. Searching for `(int)` may also find function declarations with only a single anonymous parameter.